views/dashboard.erb

```diff
@@ -0,0 +1,75 @@
+<!--padding  -->
+<div>
+  <ul>
+  </ul>
+</div>
+
+<!--add a new song  -->
+<% if current_user %>
+<div class="login-modal">
+  <div class="login-container">
+    <form method="post" action="/songs">
+      <div class=login>
+        <% if @no_song %>
+          <p> There is no song by that name. Try again.</p>
+        <% else %>
+        <p> suggest a new song</p>
+        <% end %>
+        <input id="username-login" placeholder="title" name="title"></input>
+      </div>
+    <div>
+      <input id="password-login" placeholder="artist" name="artist"></input>
+
+    </div>
+  <div>
+      <input type="submit" class="submit">
+    </form>
+  </div>
+</div>
+<% end %>
+
+<!--song titles/artist and letters  -->
+<h2><%= @songs.values.flatten.count %> songs                             </h2>
+
+<%      @songs.each do |letter, songs|                                      %>
+<h3><%=   letter                                                       %></h3>
+<%        songs.each do |song|                                              %>
+<p><%=      "Title: #{song.title}, Artist: #{song.artist}"              %></p>
+<p><iframe src="https://embed.spotify.com/?uri=spotify:track:<%="#{song.spotify_id}"%>" frameborder="0" allowtransparency="true"></iframe></p>
+<% if current_user %>
+  <%          upvote_button = 'vote-button'                                   %>
+  <%          downvote_button = 'vote-button'                                 %>
+  <%          cast_vote = current_user.votes.find {|o| o.song_id == song.id}  %>
+  <%            if cast_vote                                                  %>
+  <%              disabled_attr = 'disabled'                                  %>
+  <%              if cast_vote.value == 1                                     %>
+  <%                upvote_button = 'vote-button-down-pressed'                %>
+  <%                downvote_button = 'vote-button-down'                      %>
+  <%              else                                                        %>
+  <%                upvote_button = 'vote-button-down'                        %>
+  <%                downvote_button = 'vote-button-down-pressed'              %>
+  <%              end                                                         %>
+  <%          end                                                             %>
+  <p>
```

Comment:
As we mentioned in class, this is a great candidate for extracting a [presenter](https://robertomurray.co.uk/blog/2014/decorators-presenters-delegators-rails/) (e.g. `VoteButton.new(user).button_class_for(song)`)